compaction/compaction_manager.cc

```diff
@@ -1040,11 +1040,15 @@ class compaction_manager::offstrategy_compaction_task : public compaction_manage
     }
 };
 
-future<> compaction_manager::perform_offstrategy(replica::table* t) {
+future<bool> compaction_manager::perform_offstrategy(replica::table* t) {
     if (_state != state::enabled) {
-        return make_ready_future<>();
+        co_return false;
+    }
+    if (t->maintenance_sstable_set().all()->empty()) {
+        co_return false;
```

Comment:
I think we can possibly just allow offstrategy_compaction_task::run_offstrategy_compaction() to decide on that, which it already does. so we'd just remove this check. then we avoid this special logic for returning earlier, and leave all the decision to run_offstrategy_compaction().

It makes sense because table A was asked for off-strategy, through the trigger. Then if off-strategy isn't required we'd see:

Starting off-strategy compaction for ks.cf, N candidates were found
then
Done with off-strategy compaction for ks.cf

If there's no sstable in the maintenance set, then 0 candidates will be shown in the first message.

Tests will then be able to see when off-strategy is done on behalf of a table, even if no reshape job was needed.

Reply:
That was my first direction but for that we'll need a way to true/false from perform_task
that currently returning a void future.  It might be possible with more code, but I'm not sure it's worth it.

Reply:
done in v2